L1Trigger/L1TTrackMatch/plugins/L1TkFastVertexProducer.cc

```diff
@@ -0,0 +1,385 @@
+
+//
+// class declaration
+//
+
+class L1TkFastVertexProducer : public edm::EDProducer {
+public:
+  typedef TTTrack<Ref_Phase2TrackerDigi_> L1TTTrackType;
+  typedef std::vector<L1TTTrackType> L1TTTrackCollectionType;
+
+  explicit L1TkFastVertexProducer(const edm::ParameterSet&);
+  ~L1TkFastVertexProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginJob() override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  //virtual void beginRun(edm::Run&, edm::EventSetup const&);
+
+  // ----------member data ---------------------------
+
+  float ZMAX;    // in cm
+  float DeltaZ;  // in cm
+  float CHI2MAX;
+  float PTMINTRA;  // in GeV
+
+  float PTMAX;       // in GeV, saturation / truncation value
+  int HighPtTracks;  // saturate or truncate
+
+  int nStubsmin;    // minimum number of stubs
+  int nStubsPSmin;  // minimum number of stubs in PS modules
+
+  int nBinning;  // number of bins used in the temp histogram
+
+  bool MonteCarloVertex;  //
+                          //const StackedTrackerGeometry*                   theStackedGeometry;
+
+  bool doPtComp;
+  bool doTightChi2;
+
+  int WEIGHT;  // weight (power) of pT 0 , 1, 2
+
+  TH1F* htmp;
+  TH1F* htmp_weight;
+
+  const edm::EDGetTokenT<edm::HepMCProduct> hepmcToken;
+  const edm::EDGetTokenT<std::vector<reco::GenParticle> > genparticleToken;
+  const edm::EDGetTokenT<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > > trackToken;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+L1TkFastVertexProducer::L1TkFastVertexProducer(const edm::ParameterSet& iConfig)
+    : hepmcToken(consumes<edm::HepMCProduct>(iConfig.getParameter<edm::InputTag>("HepMCInputTag"))),
+      genparticleToken(
+          consumes<std::vector<reco::GenParticle> >(iConfig.getParameter<edm::InputTag>("GenParticleInputTag"))),
+      trackToken(consumes<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > >(
+          iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))) {
+  ZMAX = (float)iConfig.getParameter<double>("ZMAX");
+  CHI2MAX = (float)iConfig.getParameter<double>("CHI2MAX");
+  PTMINTRA = (float)iConfig.getParameter<double>("PTMINTRA");
+
+  PTMAX = (float)iConfig.getParameter<double>("PTMAX");
+  HighPtTracks = iConfig.getParameter<int>("HighPtTracks");
+
+  nStubsmin = iConfig.getParameter<int>("nStubsmin");
+  nStubsPSmin = iConfig.getParameter<int>("nStubsPSmin");
+  nBinning = iConfig.getParameter<int>("nBinning");
+
+  MonteCarloVertex = iConfig.getParameter<bool>("MonteCarloVertex");
+  doPtComp = iConfig.getParameter<bool>("doPtComp");
+  doTightChi2 = iConfig.getParameter<bool>("doTightChi2");
+
+  WEIGHT = iConfig.getParameter<int>("WEIGHT");
+
+  int nbins = nBinning;  // should be odd
+  float xmin = -30;
+  float xmax = +30;
+
+  htmp = new TH1F("htmp", ";z (cm); Tracks", nbins, xmin, xmax);
```

Comment:
these don't need to be member variables, can be declared locally inside the `produce()` function
(and can probably be declared on the stack, i.e. don't use `new `)